src/App.tsx

```diff
@@ -146,10 +146,26 @@ export default function App() {
   const [errorsDialog, setErrorsDialog] = React.useState(false);
   const dispatch = useDispatch();
   const etebase = useCredentials();
-  const darkMode = useSelector((state: store.StoreState) => state.settings.darkMode);
+  const darkModeUserSelection = useSelector((state: store.StoreState) => state.settings.darkModeUserSelection);
   const fetchCount = useSelector((state: store.StoreState) => state.fetchCount);
   const errors = useSelector((state: store.StoreState) => state.errors);
 
+  function shouldBeDark(userSelection: string): boolean {
+    if (userSelection === "auto") {
+      return window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches;
+    } else if (userSelection === "dark") {
+      return true;
+    }
+    return false;
+  }
+  const [darkMode, setDarkMode] = React.useState(() => shouldBeDark(darkModeUserSelection));
+  React.useEffect(() => {
+    setDarkMode(shouldBeDark(darkModeUserSelection));
+  }, [darkModeUserSelection]);
+  window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", () => {
```

Comment:
This listener should be added in its own `useEffect` and removed when the useEffect destructor is called.
It should also probably just set a local state variable `darkModeBrowserPrefers` or something which defaults to the current value.

Reply:
Thanks for the pointer, I didn't about this so I looked things up, I hope it's better now. It certainly looks cleaner in my opinion.

Reply:
Yup, looks great!